internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal_test.go

```diff
@@ -5,18 +5,68 @@
 package metal_test
 
 import (
+	"context"
+	"encoding/hex"
 	"fmt"
+	"math/rand"
+	"net"
+	"net/url"
 	"testing"
 
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/cosi-project/runtime/pkg/state/impl/inmem"
+	"github.com/cosi-project/runtime/pkg/state/impl/namespaced"
 	"github.com/google/uuid"
 	"github.com/stretchr/testify/assert"
 
 	"github.com/talos-systems/talos/internal/app/machined/pkg/runtime/v1alpha1/platform/metal"
+	"github.com/talos-systems/talos/pkg/machinery/nethelpers"
+	"github.com/talos-systems/talos/pkg/machinery/resources/hardware"
+	"github.com/talos-systems/talos/pkg/machinery/resources/network"
 )
 
+func generateMAC() string {
+	const macLen = 6
+
+	bytes := make([]byte, macLen)
+	if _, err := rand.Read(bytes); err != nil {
```

Comment:
I know it's a test, but doing `rand.Read()` from `math/rand` might set a bad precedent for secure random data.

I don't know if we really want a random one each time? or a fixed value would be sufficient?